src/components/Board.js

```diff
@@ -5,29 +5,108 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
 
 class Board extends Component {
   constructor() {
     super();
-
     this.state = {
       cards: [],
     };
   }
 
-  render() {
-    return (
-      <div>
-        Board
-      </div>
-    )
+  getCards = (input) => {
+    return input.data.map(inputCard => inputCard.card)
+  }
+
+  componentDidMount() {
+    const { boardsUrl, boardName, cards } = this.props
+    const retrieveListUrl = boardsUrl + boardName + cards
+
+    axios.get(retrieveListUrl)
+    .then((response) => {
+      const cards = this.getCards(response);
+      this.setState({ cards });
+    })
+    .catch((error) => {
+      this.setState({ error: error.message});
+    });
   }
 
-}
+  addCard = (newCardData) => {
+    const { boardsUrl, boardName, cards } = this.props
+    const addCardUrl = boardsUrl + boardName + cards
+
+    axios.post(addCardUrl, newCardData)
+    .then((response) => {
+      const cards = this.state.cards;
+      cards.push(response.data.card);
+      this.setState({
+        cards,
+        message: 'Added a new card successfully!'});
+      })
+      .catch((error) => {
+        this.setState({ error: error.message });
+      });
+    };
+
+    deleteCard = (card) => {
+      const { cardsUrl } = this.props
+      const cardId = card.props.id
+      const deleteCardUrl  = cardsUrl + cardId
+
+      axios.delete(deleteCardUrl)
+      .then((response) => {
+        const cards = this.state.cards;
+        const index = card.props.index
```

Comment:
I like that you've kept all the API interaction logic in one place - the callbacks are a little more complex, but I would say it makes the app as a whole much easier to comprehend. Whether or not you intended it, this is a great example of the container component pattern well-applied.